Tests/SwiftFormatWhitespaceLinterTests/WhitespaceTestCase.swift

```diff
@@ -1,54 +1,23 @@
 import SwiftFormatConfiguration
 import SwiftFormatCore
+import SwiftFormatTestSupport
 import SwiftFormatWhitespaceLinter
 import SwiftSyntax
 import XCTest
 
-/// This test class allows us to check for specific diagnostics emitted by the DiagnosticEngine. It
-/// is similar to SwiftFormatRulesTests/DiagnosingTestCase, except that this class also tests for
-/// the specific line and column numbers of the diagnostic location.
-class WhitespaceTestCase: XCTestCase {
-
-  /// The context in which to run the test.
-  private(set) var context: Context?
-
-  /// Keeps track of the linter errors queued up in the DiagnosticEngine.
-  private var consumer = DiagnosticTrackingConsumer()
-
-  /// Used to indicate that a test should fail if we don't have test assertions for all diagnotics
-  /// emmitted.
-  private var shouldCheckForUnassertedDiagnostics = false
-
-  private class DiagnosticTrackingConsumer: DiagnosticConsumer {
-    var registeredDiagnostics = [(String, line: Int?, column: Int?)]()
-
-    func handle(_ diagnostic: Diagnostic) {
-      let loc = diagnostic.location
-      registeredDiagnostics.append((diagnostic.message.text, line: loc?.line, column: loc?.column))
-    }
-
-    func finalize() {}
-  }
-
-  override func tearDown() {
-    guard shouldCheckForUnassertedDiagnostics else { return }
-
-    for diag in consumer.registeredDiagnostics {
-      XCTFail("unexpected diagnostic '\(diag)' emitted")
-    }
+class WhitespaceTestCase: DiagnosingTestCase {
+  override func setUp() {
```

Comment:
Should this call `super.setUp`? Any chance of `DiagnosingTestCase.setUp` containing some code that needs to be executed?

Reply:
Yes! Even though all we do here is change the value of `shouldCheckForUnassertedDiagnostics`, if we ever add more logic in the base class we'll want to make sure it's not skipped.